app/controllers/AlertController.java

```diff
@@ -178,16 +272,54 @@ public Result query(
         return viewAlert;
     }
 
+    private Quantity convertToInternalQuantity(final models.view.Quantity viewQuantity) {
+        if (viewQuantity == null) {
+            return null;
+        }
+        return new DefaultQuantity.Builder()
+                .setUnit(viewQuantity.getUnit())
+                .setValue(viewQuantity.getValue())
+                .build();
+    }
+
+    private NagiosExtension getNagiosExtensions(final Map<String, Object> extensionsMap) throws IOException {
+        if (extensionsMap == null || extensionsMap.isEmpty()) {
+            return null;
```

Comment:
rename: convertToInternalNagiosExtension

I am not a fan of returning null here. At the _very_ least you should mark the return result nullable. Alternatively, consider returning an optional.